### PR TITLE
Fix/network management flow fixes final

### PR DESCRIPTION
**Pull request by svojsu**
Fixed following bugs in network management flow:

86951qrg2 Undefined error on adding EVM network
86951gv3f Able to delete last custom network node
86951hd83 Coingecko url fixes
86951hmqw Invalid chain id validation
86951znzc Edit network details disable rpc url field

### CODE REVIEW

File: novawallet/Modules/NetworkManagement/NetworkNode/Interactor/NetworkNodeBaseInteractor.swift [hunk 1]
```diff
@@ -42,6 +42,31 @@ class NetworkNodeBaseInteractor: NetworkNodeConnectingTrait, NetworkNodeCorrespo
         completeSetup()
     }
 
+    func connect(
+        to node: ChainNodeModel,
+        replacing _: ChainNodeModel?,
```
Comment (ERussel) [inline novawallet/Modules/NetworkManagement/NetworkNode/Interactor/NetworkNodeBaseInteractor.swift:47]: Probably forgot to use replacing argument

File: novawallet.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -23067,7 +23067,7 @@
 			outputFileListPaths = (
 			);
 			outputPaths = (
-				"$PROJECT_DIR/NovaPushNotificationServiceExtension/R.generated.swift",
+				$PROJECT_DIR/NovaPushNotificationServiceExtension/R.generated.swift,
```
Comment (ERussel) [inline novawallet.xcodeproj/project.pbxproj:23070]: Why the quotes have been removed?

Comment (svojsu) [reply novawallet.xcodeproj/project.pbxproj:23070]: Good question. It wasn't manually for sure.

File: novawallet/Modules/NetworkManagement/NetworkDetails/NetworkDetailsPresenter.swift [hunk 3]
```diff
@@ -139,12 +139,22 @@ private extension NetworkDetailsPresenter {
...
 
     func provideNodeViewModel(for node: ChainNodeModel) {
+        let notSingleUserNode = chain.nodes
+            .filter { $0.source == .user }
+            .count > 1
+        let notEmptyRemoteModes = !chain.nodes
+            .filter { $0.source == .remote }
+            .isEmpty
+
+        let deletionAllowed = notSingleUserNode || notEmptyRemoteModes
```
[the remark this answers is not in the document]
Comment (svojsu) [reply novawallet/Modules/NetworkManagement/NetworkDetails/NetworkDetailsPresenter.swift:156]: > Didn't get this. What we are trying to do here? For example, if we have single remote node then looks like we can remove it.

We need this flag specifically for the custom nodes section. Based on this flag, we can decide whether to show "Edit" or "More options" for the node cells. If it's not a single user node or if remote nodes are present, we can delete the user nodes.

**Review by ERussel — APPROVED** [review]

**Pull request merged**